server/support-session/index.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Answers whether an inbound Express request looks like a support session
+ *
+ * @param {object} request Express request
+ *
+ * @returns {Boolean} Whether the request looks like a support session
+ */
+export function isSupportSession( request ) {
+	return !! getSupportSession( request );
+}
+
+/**
+ * Gets the support session nonce from an incoming Express request
+ *
+ * @param {object} request Express request
+ *
+ * @returns {?string} The support session nonce for a request, if any.
+ */
+export function getSupportSession( request ) {
+	return request.get( 'x-support-session' );
+}
+
+/**
+ * Sets the support session nonce for an outbound superagent request
+ *
+ * @param {object} request Superagent request
+ * @param {string} supportSession The support session nonce
+ */
+export function setSupportSession( request, supportSession ) {
+	request.set( 'x-support-session', supportSession );
```

Comment:
Yeah, that is confusing and bothered me, but I thought it would still be good to be able to read it all in a single module. What do you think about renaming the arguments to be more explicit like `superagentRequest` and `expressRequest`. If you don't like that, I'm happy to dissolve this module. It's how things started anyway. :)

Reply:
I favor simply dissolving the module 😄 Each of the functions is used exactly once. And the call-site is always a place where a low-level handling of headers is already done. The following code is therefore a bit odd:
```js
res.set( 'x-low-level-header', hmacChecksumOfSomething );
if ( req.cookies.wordpress_logged_in && req.query.foo === 'yes' ) {
  res.set( 'Authorization', `token ${ authToken }` );
}
setSupportSession( res );
```
See how the last line is on a different level of abstraction that the previous lines? A simple:
```js
req.set( 'x-support-session', supportSession );
```
is a better fit. It's all mostly personal taste based on aesthetics, I hope it makes sense anyway :wink:

Reply:
yeah, it makes sense, and I felt that mismatch when updating user-bootstrap. Dissolving the module momentarily...

Reply:
> I see it's already dissolved

It's still there and is being used in the latest version of the patch. Another rebase error again?